mesonbuild/rewriter.py

```diff
@@ -18,14 +18,15 @@
 import typing as T
 
 if T.TYPE_CHECKING:
+    from argparse import ArgumentParser, HelpFormatter
     from .mparser import BaseNode
 
 class RewriterException(MesonException):
     pass
 
 # Note: when adding arguments, please also add them to the completion
 # scripts in $MESONSRC/data/shell-completions/
-def add_arguments(parser, formatter=None):
+def add_arguments(parser: ArgumentParser, formatter: T.Callable[[str], HelpFormatter] = None) -> None:
```

Comment:
Shouldn't that be `T.Optional[Callable[...]] = None`?

Reply:
Ironically, yes. This is actually two typing related sets of changes that happened at different times and this change isn't from the ones where I was focused on None-safety.

Reply:
The None should be removed entirely.